libsolidity/codegen/AsmCodeGen.cpp

```diff
@@ -184,18 +184,20 @@ void CodeGenerator::assemble(
 )
 {
 	EthAssemblyAdapter assemblyAdapter(_assembly);
+	shared_ptr<EVMDialect> dialect = EVMDialect::strictAssemblyForEVM();
+	CodeTransform transform(
+		assemblyAdapter,
+		_analysisInfo,
+		_parsedData,
+		*dialect,
```

Comment:
Not necessarily related to this PR, since it's done all over the place - but maybe it'd be better to pass around dialects as ``shared_ptr``s - they are created as such anyways and it seems dangerous to have objects that implicitly depend on the lifetime of other objects like that...

Reply:
I agree - I just did not want to change that much code for now....